pkg/crypto.go

```diff
@@ -227,35 +230,76 @@ func (client *Crypto) StoreVendorCACertificate(certificate *x509.Certificate) er
 	return client.Storage.SaveCertificate(key, certificate.Raw)
 }
 
+func (client *Crypto) SelfSignVendorCACertificate(name string) (*x509.Certificate, error) {
+	identity := core.NutsConfig().Identity()
+	var csr *x509.CertificateRequest
+	csrAsASN1, privateKey, err := client.generateVendorCACSR(name, identity);
+	if err != nil {
+		return nil, err
+	} else {
+		if csr, err = x509.ParseCertificateRequest(csrAsASN1); err != nil {
```

Comment:
Is this block reachable if our code works as tested? (the unable to parse CSR)

Reply:
Nope. I can't even (unit) test it since our code that generates the CSR is faultless :-p